### ISSUE TITLE
Converting Scala Request to Java Request throws exception

### ISSUE THREAD

**Issue by mkurz** [playframework/playframework#9789]
Play-Scala 2.7.3

Simple `GET` route:
```
GET     /                           controllers.HomeController.index
```

Action:
```scala
  def index() = Action { implicit request: Request[AnyContent] =>
    request.asJava.body // BOOOM
    ...
  }
```

Exception:
```
play.api.http.HttpErrorHandlerExceptions$$anon$1: Execution exception[[ClassCastException: play.api.mvc.AnyContentAsEmpty$ cannot be cast to play.mvc.Http$RequestBody]]
	at play.api.http.HttpErrorHandlerExceptions$.throwableToUsefulException(HttpErrorHandler.scala:351)
	at play.api.http.DefaultHttpErrorHandler.onServerError(HttpErrorHandler.scala:267)
	at play.core.server.AkkaHttpServer$$anonfun$1.applyOrElse(AkkaHttpServer.scala:448)
	at play.core.server.AkkaHttpServer$$anonfun$1.applyOrElse(AkkaHttpServer.scala:446)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:453)
	at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
	at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:92)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:94)
	at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:92)
Caused by: java.lang.ClassCastException: play.api.mvc.AnyContentAsEmpty$ cannot be cast to play.mvc.Http$RequestBody
	at play.core.j.RequestImpl.body(JavaHelpers.scala:389)
	at controllers.HomeController.$anonfun$index$1(HomeController.scala:11)
	at scala.Function1.$anonfun$andThen$1(Function1.scala:85)
	at play.api.mvc.ActionBuilderImpl.invokeBlock(Action.scala:489)
	at play.api.mvc.ActionBuilderImpl.invokeBlock(Action.scala:487)
	at play.api.mvc.ActionBuilder$$anon$10.apply(Action.scala:426)
	at play.api.mvc.Action.$anonfun$apply$2(Action.scala:98)
	at play.api.libs.streams.StrictAccumulator.$anonfun$mapFuture$4(Accumulator.scala:184)
	at scala.util.Try$.apply(Try.scala:210)
	at play.api.libs.streams.StrictAccumulator.$anonfun$mapFuture$3(Accumulator.scala:184)
```

Problem is that Scala's `AnyContent` can not be mapped to Java's `Http.RequestBody` class.
Also [this line](https://github.com/playframework/playframework/blob/48f2b6cd1e4d2040907505d16f71d67fb2cf582d/core/play/src/main/scala/play/api/mvc/Request.scala#L82) causes a problem because the `case` clause should not be entered, but it is... `[...]` type is ignored.

**Comment by yasha-dev1** [comment]
Is there any update on this?

**Comment by mkurz** [comment]
@yasha-dev1 Fixed in #11930 in Play 2.9.
However please be aware of this comment:
https://github.com/playframework/playframework/blob/9fb5cc8dce16403859170f7e49da00581a9faff2/core/play/src/main/scala/play/api/mvc/Request.scala#L111-L120
